src/media/media-upload.entity.ts

```diff
@@ -59,7 +59,7 @@ export class MediaUpload {
     extension: string,
     backendType: BackendType,
     backendData?: string,
-  ): MediaUpload {
+  ): Omit<MediaUpload, 'createdAt'> {
```

Comment:
I think this entity has to be modified. Can the Note be null or not? What is with `fileUrl`? It is never set in the create method.
It would be nicer if the `backendData` has the default value `null`.

Reply:
`note` is nullable in the schema, "as it is possible to delete a note without also deleting the associated media uploads".
But when *creating* a new upload, a note must be provided. This fact should probably be mentioned in the docs of this method, though.

Having no `fileUrl` does not seem to make sense at first glance.

To be consistent with the other methods, `backendData` should not be part of this `create`.

Reply:
The note can be `null`, if the note is deleted afterwards and the mediaUploads still persists. `backendData` can be either `null` or a `string` depending on the `backendType`. `fileUrl` should always be a `string`, but we're in the awkward position that we need the `id` of the upload to get the `fileUrl` from the appropiate upload handler… So I'm not sure what to do here…